cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -343,6 +353,51 @@ impl LockedManifestCatalog {
         Ok(lockfile)
     }
 
+    /// Given locked packages and manifest options allows, verify that the
+    /// locked packages are allowed.
+    fn check_packages_are_allowed(
+        locked_packages: &[LockedPackageCatalog],
+        allow: &Allows,
+    ) -> Result<(), LockedManifestError> {
+        if !allow.licenses.is_empty() {
+            for package in locked_packages {
+                if let Some(license) = &package.license {
+                    if !allow.licenses.contains(license) {
+                        return Err(LockedManifestError::LicenseNotAllowed(
+                            package.install_id.clone(),
+                            license.clone(),
+                        ));
+                    }
+                }
+            }
+        }
```

Comment:
nit: i think the pkgdb implementation collected _a list_ of warning/errors to report on multiple packages at once.
It's easyish to add that back in the future so if were time pressed thats something to defer.

Reply:
pkgdb throws for errors and collects a list of warnings, which is the same as what this patch does
https://github.com/flox/flox/blob/ee3772f83c79aa249fee892de9787e244dca6033/pkgdb/src/resolver/lockfile.cc#L352